tests/cnf/ran/oran/tests/oran-pre-provision.go

```diff
@@ -1 +1,141 @@
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/oran/internal/helper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/oran/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Some test cases are marked as pending from the code. This is used for tests that cannot be used with the loopback
+// adaptor and are guaranteed to fail until the production-ready DTIAS is available.
+
+var _ = Describe("ORAN Pre-provision Tests", Label(tsparams.LabelPreProvision), func() {
+	// 77386 - Failed authentication with hardware manager
+	PIt("fails to authenticate with hardware manager", reportxml.ID("77386"), func() {
+		By("getting a valid Dell HardwareManager")
+		hwmgr, err := helper.GetValidDellHwmgr(HubAPIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get a valid Dell HardwareManager")
+
+		By("getting the Dell AuthSecret")
+		authSecret, err := secret.Pull(
+			HubAPIClient, hwmgr.Definition.Spec.DellData.AuthSecret, tsparams.HardwareManagerNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get the HardwareManager AuthSecret")
+		Expect(authSecret.Definition.Data).ToNot(BeNil(), "HardwareManager AuthSecret must have data")
+
+		By("copying the secret and updating the password")
+		authSecret.Definition.Name += "-test"
+		authSecret.Definition.Data["password"] = []byte(tsparams.TestBase64Credential) // wrongpassword
+
+		authSecret, err = authSecret.Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create the new AuthSecret")
+
+		By("copying the HardwareManager and updating the AuthSecret")
+		hwmgr.Definition.Name += "-test"
+		hwmgr.Definition.Spec.DellData.AuthSecret = authSecret.Definition.Name
+
+		hwmgr, err = hwmgr.Create()
+		Expect(err).ToNot(HaveOccurred(), "Failed to create the new HardwareManager")
+
+		By("waiting for the authentication to fail")
+		hwmgr, err = hwmgr.WaitForCondition(tsparams.HwmgrFailedAuthCondition, time.Minute)
+		Expect(err).ToNot(HaveOccurred(), "Failed to wait for the HardwareManager to fail authentication")
+
+		By("deleting the invalid HardwareManager")
+		err = hwmgr.Delete()
+		Expect(err).ToNot(HaveOccurred(), "Failed to delete the invalid HardwareManager")
+
+		By("deleting the invalid AuthSecret")
+		err = authSecret.Delete()
+		Expect(err).ToNot(HaveOccurred(), "Failed to delete the invalid AuthSecret")
+	})
+
+	// 77392 - Apply a ProvisioningRequest referencing an invalid ClusterTemplate
+	It("fails to create ProvisioningRequest with invalid ClusterTemplate", reportxml.ID("77392"), func() {
+		By("attempting to create a ProvisioningRequest")
+		prBuilder := helper.NewProvisioningRequest(HubAPIClient, tsparams.TemplateInvalid)
+		_, err := prBuilder.Create()
+		Expect(err).To(HaveOccurred(), "Creating a ProvisioningRequest with an invalid ClusterTemplate should fail")
+	})
+
+	DescribeTable("ClusterTemplate failed validations",
+		func(templateVersion, message string) {
+			By("verifying the ClusterTemplate validation failed")
```

Comment:
nit: `By("verifying the ClusterTemplate validation failed with proper message")`